### PR TITLE
CORS Following (issue while making request with axios)

### PR DESCRIPTION
**Pull request by Unkodero**
https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS

### CODE REVIEW

File: src/OAuth2/Controller/TokenController.php [hunk 1]
```diff
@@ -97,9 +97,15 @@ public function handleTokenRequest(RequestInterface $request, ResponseInterface
...
     {
+        if (strtolower($request->server('REQUEST_METHOD')) == 'options') {
+            $response->addHttpHeaders(array('Allow' => 'POST, OPTIONS'));
+
+            return null;
+        }
+        
         if (strtolower($request->server('REQUEST_METHOD')) != 'post') {
             $response->setError(405, 'invalid_request', 'The request method must be POST when requesting an access token', '#section-3.2');
-            $response->addHttpHeaders(array('Allow' => 'POST'));
```
Comment (ApacheEx) [inline src/OAuth2/Controller/TokenController.php]: what is the reason of deletion this line? it should be here. Everything else looks good.

Comment (bshaffer) [inline src/OAuth2/Controller/TokenController.php]: Instead, this should be `array('Allow' => array('POST', 'OPTIONS'))`

**Comment by bshaffer** [comment]
Can we get a Test Case added for this?

File: src/OAuth2/Controller/TokenController.php (hunk 1 above)
Comment (ApacheEx) [inline src/OAuth2/Controller/TokenController.php]: Yeah, that's true

**Comment by ApacheEx** [comment]
@bshaffer,
I've added test cases. Please check #829

**Comment by bshaffer** [comment]
This is great! #829 is merged

**Pull request closed without merging**